### PR TITLE
quick: Add yaml format support

### PR DESCRIPTION
**Pull request by vadmeste**
## Description
quick Save() and Load() infers config file's format from
file name extension.

## Motivation and Context
Switching to yaml config file format

## How Has This Been Tested?
go test + Manually

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

### CODE REVIEW

**Review by harshavardhana — COMMENTED** [review]
Also add another helper function to migrate from json to yaml, given that we make sure json as "Version" string we can do the same for yaml.  

This should be done transparently without usage of any structs along the way.

**Review by harshavardhana — COMMENTED** [review]
An example like 

```yml
---
version: '14'
credential:
  accessKey: USWUXHGYZQYFYFFIT3RE
  secretKey: MOJRH0mkL1IPauahWITSVvyDrQbEEIwljvmxdq03
region: us-east-1
browser: 'on'
logger:
  console:
    enable: true
    level: error
  file:
    enable: false
    fileName: test.log
    level: error
notify:
  amqp:
    '2':
      enable: false
      url: amqp://guest:guest@localhost:5672/
      exchange: minio
      routingKey: minio
      exchangeType: direct
      mandatory: false
      immediate: false
      durable: false
      internal: false
      noWait: false
      autoDeleted: false
  nats:
    '1':
      enable: false
      address: ''
      subject: ''
      username: ''
      password: ''
      token: ''
      secure: false
      pingInterval: 0
      streaming:
        enable: false
        clusterID: ''
        clientID: ''
        async: false
        maxPubAcksInflight: 0
  elasticsearch:
    '1':
      enable: false
      url: http://localhost:9200
      index: minioevents
  redis:
    '1':
      enable: false
      address: 127.0.0.1:6379
      password: ''
      key: ''
  postgresql:
    '1':
      enable: false
      connectionString: ''
      table: ''
      host: ''
      port: ''
      user: ''
      password: ''
      database: ''
  kafka:
    '1':
      enable: false
      brokers: 
      topic: ''
  webhook:
    '1':
      enable: false
      endpoint: ''
```

```js
{
	"version": "14",
	"credential": {
		"accessKey": "USWUXHGYZQYFYFFIT3RE",
		"secretKey": "MOJRH0mkL1IPauahWITSVvyDrQbEEIwljvmxdq03"
	},
	"region": "us-east-1",
	"browser": "on",
	"logger": {
		"console": {
			"enable": true,
			"level": "error"
		},
		"file": {
			"enable": false,
			"fileName": "test.log",
			"level": "error"
		}
	},
	"notify": {
		"amqp": {
			"2": {
				"enable": false,
				"url": "amqp://guest:guest@localhost:5672/",
				"exchange": "minio",
				"routingKey": "minio",
				"exchangeType": "direct",
				"mandatory": false,
				"immediate": false,
				"durable": false,
				"internal": false,
				"noWait": false,
				"autoDeleted": false
			}
		},
		"nats": {
			"1": {
				"enable": false,
				"address": "",
				"subject": "",
				"username": "",
				"password": "",
				"token": "",
				"secure": false,
				"pingInterval": 0,
				"streaming": {
					"enable": false,
					"clusterID": "",
					"clientID": "",
					"async": false,
					"maxPubAcksInflight": 0
				}
			}
		},
		"elasticsearch": {
			"1": {
				"enable": false,
				"url": "http://localhost:9200",
				"index": "minioevents"
			}
		},
		"redis": {
			"1": {
				"enable": false,
				"address": "127.0.0.1:6379",
				"password": "",
				"key": ""
			}
		},
		"postgresql": {
			"1": {
				"enable": false,
				"connectionString": "",
				"table": "",
				"host": "",
				"port": "",
				"user": "",
				"password": "",
				"database": ""
			}
		},
		"kafka": {
			"1": {
				"enable": false,
				"brokers": null,
				"topic": ""
			}
		},
		"webhook": {
			"1": {
				"enable": false,
				"endpoint": ""
			}
		}
	}
}
```

File: pkg/quick/config_encoding.go [hunk 1]
```diff
@@ -0,0 +1,143 @@
+/*
+ * Quick - Quick key value store for config files and persistent state files
+ *
+ * Minio Client (C) 2015 Minio, Inc.
```
Comment (harshavardhana) [inline pkg/quick/config_encoding.go]: Fix the copyRight header.

File: pkg/quick/config_encoding.go [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+	if err != nil {
+		return err
+	}
+	fileData, err := ioutil.ReadFile(filename)
+	if err != nil {
+		return err
+	}
+	if runtime.GOOS == "windows" {
+		fileData = []byte(strings.Replace(string(fileData), "\r\n", "\n", -1))
+
```
Comment (harshavardhana) [inline pkg/quick/config_encoding.go]: Unnecessary space

**Review by harshavardhana — CHANGES_REQUESTED** [review]
Windows build failed..

**Comment by vadmeste** [comment]
> Also add another helper function to migrate from json to yaml, given that we make sure json as "Version" string we can do the same for yaml.
> 
> This should be done transparently without usage of any structs along the way.

Migrating is easy with quick now: quick.Load("config.json") then quick.Save("config.yaml")

**Comment by vadmeste** [comment]
> Migrating is easy with quick now: quick.Load("config.json") then quick.Save("config.yaml")

I understand it is better without structs, but we don't need it now I guess, right?

**Comment by harshavardhana** [comment]
> Migrating is easy with quick now: quick.Load("config.json") then quick.Save("config.yaml")

True but was thinking that we can have a helper in quick to translate between json to yaml and vice versa.

File: pkg/quick/config_encoding.go [hunk 3]
```diff
@@ -0,0 +1,142 @@
+/*
+ * Quick - Quick key value store for config files and persistent state files
+ *
+ * Minio Client (C) 2017 Minio, Inc.
```
Comment (harshavardhana) [inline pkg/quick/config_encoding.go]: shouldnt be minio client..

File: pkg/quick/config_encoding.go [hunk 4]
```diff
@@ -0,0 +1,142 @@
+/*
```
Comment (harshavardhana) [inline pkg/quick/config_encoding.go]: rename the file as just encoding.go

**Review by harshavardhana — CHANGES_REQUESTED** [review]

**Review by harshavardhana — APPROVED** [review]

**Pull request merged**